### ISSUE TITLE
Send python lines to ipython under Windows

### ISSUE THREAD

**Issue by harryln** [randy3k/Terminus#89]
Dear Sir,
I use `SendCode` to send python lines to `iPython` lunched in Terminus under **Windows 7/10**. One thing is really annoyed me that the **line feed/carriage return** cannot be rendered in `IPython`. I have to jump into `IPython` to press RETURN.  If using `python` instead of `IPython`, this is fine. Any idea or suggestion about this? I guess that this may be the problem of python package `prompt-toolkit` used by `IPython`.  By the way for Linux, it works perfectly.

**Comment by randy3k** [comment]
I remember I have fixed a bug related to ipython use on Terminus and Windows. What version of ipython was it?

**Comment by harryln** [comment]
The ipython version is 7.2.0. Thanks very much.

**Comment by harryln** [comment]
Yes. Indeed it worked for old ipython. But after updating ipython, it does not work.

**Comment by randy3k** [comment]
Could you confirm that with the new version of ipython you could still send (with CR) a single line but not multiple lines don't work?

**Comment by harryln** [comment]
Hi, Please see the following:

The multiple lines:
```
In [5]: import numpy as np                                                              
   ...: from sklearn import metrics                                                     
   ...: y = np.array([1, 1, 2, 2])                                                      
   ...: pred = np.array([0.1, 0.4, 0.35, 0.8])                                          
   ...: fpr, tpr, thresholds = metrics.roc_curve(y, pred, pos_label=2)                  
   ...: metrics.auc(fpr, tpr)^[                                                         
   ...:                                                                                 
```

The single line:
```
In [3]: alist = ['a1', 'a2', 'a3']                                                      
   ...:                                                                                 
   ...:     
```

You can see there is an extra `^[` for the multiple lines. There is no RETURN for both.

**Comment by randy3k** [comment]
I think I have the fix: da16b82f92f8a179379b08437f7f2da218c23612

It would be great if you could check out the master branch and test it.

**Comment by harryln** [comment]
The results:
```
In [12]: alist = ['a1', 'a2', 'a3']                                                     
                                                                                        
In [13]: blist = ['b1', 'b2', 'b3']                                                     
                                                                                        
In [14]: for i, a in enumerate(alist):                                                  
    ...:     print( i, a )^[                                                            
    ...:                                                                                
```
The single line is ok but block is not.

**Comment by randy3k** [comment]
I may need to find a clever trick. I also verify that it was working with ipython 6

**Comment by harryln** [comment]
Many thanks for your great efforts.

**Comment by randy3k** [comment]
could you try again?

**Comment by randy3k** [comment]
In ipython, the key `<esc>+<enter>` was used to commit multiple lines to the console after pasting the lines to the console. However for some reasons, the new version of ipython has a very strange restriction of the delays between the `<esc>` and the `<enter>` key. Basically, the `<enter>` key cannot be sent immediately after the `<esc>` key, but the interval also cannot be too long.

I am still looking for a cleaner solution, adding a delay doesn't always work.

**Comment by randy3k** [comment]
Okay, I found an alternative solution. Try the master of SendCode again.

**Comment by harryln** [comment]
Dear @randy3k, it works perfectly for both single and multiple line. Thank you very much for your great efforts and wonderful tools such as SendCode and Terminus!
